tests/application_development/code_relocation/src/test_file1.c

```diff
@@ -20,25 +20,66 @@ void function_in_custom_section(void);
 
 ZTEST(code_relocation, test_function_in_sram2)
 {
+	extern int __sram2_data_start;
+	extern int __sram2_data_end;
+	extern int __sram2_bss_start;
+	extern int __sram2_bss_end;
+	extern int __sram2_rodata_start;
+	extern int __sram2_rodata_end;
+	extern int __sram_text_start;
+	extern int __sram_text_end;
+	extern int __custom_section_start;
+	extern int __custom_section_end;
+	int sram2_data_start = (int)&__sram2_data_start;
```

Comment:
2 things (but I will take a better look later):
1. this should be `extern char foobar[];`
2. do not cast to `int`, use `uint32_t` or `uintptr_t`

Also, do not create a redundant variable, just use the `__*` variables directly.